### PR TITLE
feat: daily check-in enrolment persistence (#109)

### LINKED ISSUE #109: Action Server: Daily CI enrollment persistence in DB

### PR DESCRIPTION
**Pull request by mathieubergeron**
## Description
- Daily ci enrollment are now persisted within DB
- Failure to save reminder within the DB results in error message within the conversation

## Related issues
closes #109 

## Checklist

- [x] PR title follows [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) guidelines 
- [x] All relevant PR sections are populated, irrelevant ones are removed

### CODE REVIEW

File: action-server/tests/test_daily_ci_enroll_form.py [hunk 1]
```diff
@@ -641,3 +664,45 @@ def test_provide_has_dialogue_deny(self):
...
+        self.assert_templates(
+            [
+                "utter_daily_ci_enroll__enroll_fail_1",
+                "utter_daily_ci_enroll__enroll_fail_2",
+                "utter_daily_ci_enroll__enroll_fail_3",
+            ]
+        )
+
+        expected_reminder = Reminder(None)
+        expected_reminder.pre_existing_conditions = True
```
Comment (LaurenceDupont) [inline action-server/tests/test_daily_ci_enroll_form.py:703]: I renamed this to preconditions to match the spec.

File: action-server/db/reminder.py [hunk 2]
```diff
@@ -23,18 +71,103 @@ class Reminder(Base):
...
+    def pre_existing_conditions(self, pre_existing_conditions):
+        self._set_attribute(PRE_EXISTING_CONDITIONS_ATTRIBUTE, pre_existing_conditions)
+
+    @property
+    def has_dialogue(self):
+        return self.attributes[HAS_DIALOGUE_ATTRIBUTE]
+
+    @has_dialogue.setter
+    def has_dialogue(self, has_dialogue):
+        self._set_attribute(HAS_DIALOGUE_ATTRIBUTE, has_dialogue)
```
Comment (LaurenceDupont) [inline action-server/db/reminder.py:173]: We try to use type annotations elsewhere for both parameters and return types. Can you add these?

File: action-server/actions/daily_ci_enroll_form.py [hunk 3]
```diff
@@ -270,14 +275,41 @@ def submit(
...
+        else:
+            dispatcher.utter_message(template="utter_daily_ci_enroll__enroll_fail_1")
+            dispatcher.utter_message(template="utter_daily_ci_enroll__enroll_fail_2")
+            dispatcher.utter_message(template="utter_daily_ci_enroll__enroll_fail_3")
 
         return []
 
 
+def _save_reminder(reminder):
+    if reminder.phone_number[4:7] == "555":
```
Comment (LaurenceDupont) [inline action-server/actions/daily_ci_enroll_form.py:296]: What is the allowed phone number format? At some point, we allowed both 10 and 11 digits. Not sure if that's still the case. We just have to make sure that the indexation above matches the allowed format, whether it's 10 or 11 digits.

File: action-server/actions/daily_ci_enroll_form.py [hunk 4]
```diff
@@ -270,14 +275,41 @@ def submit(
         tracker: Tracker,
         domain: Dict[Text, Any],
     ) -> List[Dict]:
-        if tracker.get_slot(DO_ENROLL_SLOT) is True:
+        if tracker.get_slot(DO_ENROLL_SLOT) is False:
```
Comment (LaurenceDupont) [inline action-server/actions/daily_ci_enroll_form.py:278]: Is this slot initialized to False? I just want to make sure that when we reach this point the slot can't be none.

Comment (mathieubergeron) [reply action-server/actions/daily_ci_enroll_form.py:278]: good point!

File: action-server/actions/daily_ci_enroll_form.py (hunk 3 above)
Comment (mathieubergeron) [reply action-server/actions/daily_ci_enroll_form.py:296]: Indeed, I should assert the length. It should be exactly 11.

File: action-server/db/reminder.py (hunk 2 above)
Comment (mathieubergeron) [reply action-server/db/reminder.py:173]: will do!

File: action-server/tests/test_daily_ci_enroll_form.py (hunk 1 above)
Comment (mathieubergeron) [reply action-server/tests/test_daily_ci_enroll_form.py:703]: fixed

File: action-server/db/reminder.py [hunk 5]
```diff
@@ -1,11 +1,59 @@
...
+
+FIRST_NAME_SLOT = "first_name"
+PHONE_NUMBER_SLOT = "phone_number"
+LANGUAGE_SLOT = "language"
+PROVINCE_SLOT = "province"
+AGE_OVER_65_SLOT = "age_over_65"
+PRECONDITIONS_SLOT = "preconditions"
+HAS_DIALOGUE_SLOT = "has_dialogue"
+METADATA_SLOT = "metadata"
+TIMEZONE_METADATA_PROPERTY = "timezone"
```
Comment (mathieubergeron) [inline action-server/db/reminder.py:29]: @LaurenceDupont new slots constants that will need to be extracted in your branch

**Pull request merged**